### PR TITLE
Fix detection of sqlite on macOS

### PR DESCRIPTION
**Pull request by danpat**
# Issue

On macOS, the system-provided `libsqlite` has extension-loading disabled (for security reasons), which means it can't be used with `libspatialite`.

The fix is to `brew install sqlite3`, which puts a usable copy of sqlite in `/usr/local/opt/sqlite3`.

This PR updates `cmake/FindSQlite3.cmake` for macOS so it:

  - prefers /usr/local/opt/sqlite3 when it exists
  - fails when the `sqlite3_enable_load_extension` function is missing from whatever `libsqlite.a/.dylib` it finds

and it'll suggest to the user that they `brew install sqlite3` to fix the problem.

## Tasklist

 - [ ] review - you must request approval to merge any PR to master
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Add to release notes in valhalla-docs
 - [ ] update relevant documentation if this is an API impacting change

### CODE REVIEW

File: cmake/FindSQLite3.cmake [hunk 1]
```diff
@@ -61,9 +89,18 @@ find_package_handle_standard_args(SQLITE3
 
 mark_as_advanced(SQLITE3_LIBRARY SQLITE3_INCLUDE_DIR SQLITE3_LIBRARIES)
 
-if(NOT SQLite3_FIND_QUIETLY)
+if(NOT SQLITE3_FIND_QUIETLY)
```
Comment (oxidase) [inline cmake/FindSQLite3.cmake]: why this change? `SQLite3_FIND_QUIETLY` is correct variable name for the  `QUIET` option of `SQLite3` package.

Comment (danpat) [reply cmake/FindSQLite3.cmake]: This was a search/replace typo that slipped in.  Fixed, thanks for spotting it.
